server/src/main/java/org/opensearch/common/settings/Setting.java

```diff
@@ -2073,6 +2183,30 @@ public static Setting<TimeValue> timeSetting(
         );
     }
 
+    private static class TimeValueParser implements Function<String, TimeValue>, Writeable {
+        private String key;
+        private TimeValue minValue;
+        private boolean isFiltered;
+
+        public TimeValueParser(String key, TimeValue minValue, boolean isFiltered) {
+            this.key = key;
+            this.minValue = minValue;
+            this.isFiltered = isFiltered;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeString(key);
+            out.writeTimeValue(minValue);
+            out.writeBoolean(isFiltered);
+        }
+
+        @Override
+        public TimeValue apply(String s) {
+            return minTimeValueParser(key, minValue, isFiltered).apply(s);
```

Comment:
Since you're only replacing the `minTimeValueParser` here your class name should probably match, and you should create another writeable `MinMaxTimeValueParser` for the other implementation.